locallib.php

```diff
@@ -443,21 +443,51 @@ function local_boostnavigation_cohort_is_member($userid, $setting) {
     // Initialize variable for memberships.
     static $allmemberships = null;
 
-    // First: If the memberships haven't been fetched yet initially, fetch all of the user's cohort memberships
-    // only once and remember for next calls of this function.
-    if ($allmemberships == null) {
-        // Prepare SQL statement.
-        $sql = 'SELECT {cohort}.id, {cohort}.idnumber FROM {cohort_members}
-            JOIN {cohort}
-                ON {cohort}.id = {cohort_members}.cohortid
-            WHERE {cohort_members}.userid = ?';
-        $params = array($userid);
-
-        // Run DB query.
-        $ret = $DB->get_records_sql_menu($sql, $params);
-
-        // Remember memberships.
-        $allmemberships = $ret;
+    try {
+        $cohortversioncache = cache::make('local_boostnavigation', 'cohortversion');
+        $usercohortversioncache = cache::make('local_boostnavigation', 'usercohortversion');
+        $usercohortscache = cache::make('local_boostnavigation', 'usercohorts');
+    } catch (Exception $e) {
```

Comment:
Why not just let the exceptions get thrown naturally?

Reply:
Thank you Brendan !!

Removed try catch block. Please review.

Thanks,
Anu